### PR TITLE
Perfetto profiling

### PR DESCRIPTION
**Pull request by therault**
Add a simple converter from PaRSEC Binary Profile files into a Perfetto trace file.

See https://ui.perfetto.dev/ to visualize such traces.

### CODE REVIEW

File: tools/profiling/dbp2pftrace.cc [hunk 1]
```diff
@@ -0,0 +1,327 @@
+//
+// Created by herault on 4/19/22.
```
Comment (devreal) [inline tools/profiling/dbp2pftrace.cc:2]: Should there be a proper copyright header?

File: tools/profiling/CMakeLists.txt [hunk 2]
```diff
@@ -41,5 +41,42 @@ if(Graphviz_FOUND)
 
 endif(Graphviz_FOUND)
 
-Add_Subdirectory(python)
+if(CMAKE_CXX_COMPILER)
```
Comment (bosilca) [inline tools/profiling/CMakeLists.txt:44]: The perfetto content seems to be downloaded in our source directory, and will confuse git and possibly us with new untracked files. Maybe we should have a 3rd-party directory where we keep our external dependencies.

File: tools/profiling/dbp2pftrace.cc [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+void StopTracing(std::unique_ptr<perfetto::TracingSession> tracing_session, int fd)
+{
+    // Make sure the last event is closed for this example.
+    perfetto::TrackEvent::Flush();
+
+    // Stop tracing and read the trace data.
+    tracing_session->StopBlocking();
+    std::vector<char> trace_data(tracing_session->ReadTraceBlocking());
+
+    // Write the result into a file.
```
Comment (bosilca) [inline tools/profiling/dbp2pftrace.cc:84]: Are these leftover comments from their example ?

File: tools/profiling/dbp2pftrace.cc [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+}
+
+void StopTracing(std::unique_ptr<perfetto::TracingSession> tracing_session, int fd)
+{
+    // Make sure the last event is closed for this example.
+    perfetto::TrackEvent::Flush();
+
+    // Stop tracing and read the trace data.
+    tracing_session->StopBlocking();
+    std::vector<char> trace_data(tracing_session->ReadTraceBlocking());
```
Comment (bosilca) [inline tools/profiling/dbp2pftrace.cc:82]: Why is there a read of the events ?

File: tools/profiling/CMakeLists.txt (hunk 2 above)
Comment (therault) [reply tools/profiling/CMakeLists.txt:44]: I have tested again, FetchContent_Declare / FetchContent_Populate download and expand the tar.gz file into the `${PROJECT_BINARY_DIR}/_deps/perfetto_source-src/` directory. 

The call to `  FetchContent_GetProperties(perfetto_source SOURCE_DIR PERFETTO_SOURCE_DIR)` a few lines after that allow us to find exactly where it is, but when I tested it is always in the build directory

**Comment by therault** [comment]
The Perfetto SDK is a *tracing* interface: there is no way to guarantee, when converting a file into another, that all events will be kept in the new trace. I have updated this branch to a more recent Perfetto while trying to make it work and understanding the behavior of the tracing library, this code might be re-usable if we create a new MCA framework to have alternative tracing mechanisms.

Meanwhile, the trace conversion approach needs to either go one level lower, at the Protobuf level, or to keep generating a JSON file.

**Pull request closed without merging**